nmrglue/fileio/bruker.py

```diff
@@ -2121,28 +2122,36 @@ def read_jcamp(filename):
     """
     dic = {"_coreheader": [], "_comments": []}  # create empty dictionary
 
-    with open(filename, 'r') as f:
-        while True:     # loop until end of file is found
-
-            line = f.readline().rstrip()    # read a line
-            if line == '':      # end of file found
-                break
-
-            if line[:6] == "##END=":
-                # print("End of file")
-                break
-            elif line[:2] == "$$":
-                dic["_comments"].append(line)
-            elif line[:2] == "##" and line[2] != "$":
-                dic["_coreheader"].append(line)
-            elif line[:3] == "##$":
-                try:
-                    key, value = parse_jcamp_line(line, f)
-                    dic[key] = value
-                except:
-                    warn("Unable to correctly parse line:" + line)
+    for N, encoding in enumerate(encodings):
+        try:
+            with open(filename, 'r', encoding = encoding) as f:
```

Comment:
In Python 2 `encoding` is not an argument in the buildin `open` command.  `io.open` does support this argument.